File: pegasusapp/src/models/asistencia.ts
import { Curso } from './curso';
import { Materia } from './materia';
import { Usuario } from './usuario';

export interface Asistencia {
  id: number;
  id_Alumno?: number | null;
  id_Materia?: number | null;
  id_Curso?: number | null;
  fecha?: string | null; // ISO 8601 string format, e.g., "2025-05-19T00:00:00Z"
  presente: boolean;

  alumno?: Usuario | null;
  materia?: Materia | null;
  curso?: Curso | null;
}

File: PegasusV1/PegasusV1/Entities/Asistencia.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace PegasusV1.Entities
{
    public class Asistencia
    {
        [Key]
        public int Id { get; set; }

        [ForeignKey("Id_Alumno")]
        public virtual Usuario? Alumno { get; set; }
        public int? Id_Alumno { get; set; }

        [ForeignKey("Id_Materia")]
        public virtual Materia? Materia { get; set; }
        public int? Id_Materia { get; set; }

        public int? Id_Curso { get; set; }
        [ForeignKey("Id_Curso")]
        public Curso? Curso { get; set; }

        public DateTime? Fecha { get; set; }    

        public bool? Presente { get; set; }
    }
}
